src/Microsoft.AspNet.Security/Infrastructure/AuthenticationHandler.cs

```diff
@@ -77,9 +77,12 @@ internal AuthenticationOptions BaseOptions
             if (BaseOptions.AuthenticationMode == AuthenticationMode.Active)
             {
                 AuthenticationTicket ticket = await AuthenticateAsync();
-                if (ticket != null && ticket.Identity != null)
+                if (ticket != null)
                 {
-                    SecurityHelper.AddUserIdentity(Context, ticket.Identity);
+                    if ( ticket.Identity != null)
+                        SecurityHelper.AddUserIdentity(Context, ticket.Identity);
```

Comment:
@blowdart - ClaimsPrincipal was introduced in `AuthenticationTicket` with OIDC - https://github.com/aspnet/Security/commit/49e66f03110b486f8dd9959dd2cee802d676c28f. @blowdart mentioned in a conversation having both `Principal` and `Identity` on `AuthenticationTicket` is going to confuse users. So until we decide on whether it is `Principal` or `Identity` to be in `AuthenticationTicket` he suggested to remove the `Principal` property from `AuthenticationTicket` for `beta3`. 

@brentschmaltz @Eilon @Tratcher